### PR TITLE
*: fix some golint and vet warnings, so that `make check` passes

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

`make check` was failing. This PR fixes this.

### What is changed and how it works?

Apply the vet suggestions.

Also, split off some part of `(*Syncer).Run()` into its own function so that the `declaration of "err" shadows declaration at line XXXX` warning is easier to fix. Eventually this needs to be further refactored as some sub function is still >100 lines long.

Removed some useless `errors.Trace()` calls to simplify the extracted function.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -1132,580 +1127,627 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
...
+				&currentPos,
+				&lastPos,
+				shardingReSync,
+				&latestOp,
+				closeShardingSyncer,
+				traceSource,
+				safeMode,
+				tryReSync,
+			)
 			log.Infof("rotate binlog to %v", currentPos)
```
Comment (csuzhangxc) [inline syncer/syncer.go]: move `log.Infof("rotate binlog to %v", currentPos)` into `handleRotateEvent`?

File: syncer/syncer.go [hunk 2]
```diff
@@ -1132,580 +1127,627 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
...
+	for i := range sqls {
+		var arg []interface{}
+		var key []string
+		if args != nil {
+			arg = args[i]
+		}
+		if keys != nil {
+			key = keys[i]
+		}
+		err = s.commitJob(insert, originSchema, originTable, table.schema, table.name, sqls[i], arg, key, true, *lastPos, *currentPos, nil, *traceID)
```
Comment (csuzhangxc) [inline syncer/syncer.go]:
```suggestion
		err = s.commitJob(*latestOp, originSchema, originTable, table.schema, table.name, sqls[i], arg, key, true, *lastPos, *currentPos, nil, *traceID)
```
?
can add a `default` case to `return`?

**Comment by csuzhangxc** [comment]
@WangXiangUSTC PTAL

**Comment by GregoryIan** [comment]
@amyangfei PTAL

File: syncer/syncer.go [hunk 3]
```diff
@@ -1132,580 +1127,632 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
...
-				continue
-			}
-			latestOp = rotate
-
-			if s.tracer.Enable() {
-				_, err := s.tracer.CollectSyncerBinlogEvent(traceSource, safeMode.Enable(), tryReSync, lastPos, currentPos, int32(e.Header.EventType), int32(latestOp))
-				if err != nil {
-					log.Infof("[syncer] tracer collect syncer binlog error: %v", errors.ErrorStack(err))
-				}
+			s.handleRotateEvent(
```
Comment (amyangfei) [inline syncer/syncer.go]: can we define a struct to hold all parameters of these three event handler (handleRotateEvent, handleRowsEvent and handleQueryEvent), most of parameters of these three functions are same.

Comment (GregoryIan) [reply syncer/syncer.go]: agree! Rest also LGTM

Comment (amyangfei) [reply syncer/syncer.go]: ping @kennytm

**Comment by kennytm** [comment]
@amyangfei Should I just retract the changes to `syncer.go` in favor of #147?

**Comment by amyangfei** [comment]
> @amyangfei Should I just retract the changes to `syncer.go` in favor of #147?

yep I think this pr doesn't change core logic of syncer module, It is ok to apply the changes

**Comment by GregoryIan** [comment]
CI was broken

**Comment by amyangfei** [comment]
> CI was broken

```
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (111)
Failed to start TiDB
```
seems problem caused by Jenkins CI

**Pull request merged**